File: src/main/java/com/vaadin/demo/component/tabs/TabsHorizontal.java
package com.vaadin.demo.component.tabs;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.tabs.Tab;
import com.vaadin.flow.component.tabs.Tabs;
import com.vaadin.flow.router.Route;

@Route("tabs-horizontal")
public class TabsHorizontal extends Div {

	public TabsHorizontal() {
		// tag::snippet[]
		Tab analytics = new Tab("Analytics");
		Tab customers = new Tab("Customers");
		Tab dashboards = new Tab("Dashboards");
		Tab documents = new Tab("Documents");
		Tab orders = new Tab("Orders");

		Tabs tabs = new Tabs(analytics, customers, dashboards, documents, orders);
		tabs.setMaxWidth("100%");
		tabs.setWidth("400px");
		// end::snippet[]
		add(tabs);
	}

	public static class Exporter extends DemoExporter<TabsHorizontal> { // hidden-source-line
	} // hidden-source-line
}

File: frontend/demo/component/tabs/tabs-horizontal.ts
import 'Frontend/demo/init'; // hidden-source-line

import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/vaadin-tabs/vaadin-tabs';

@customElement('tabs-horizontal')
export class Example extends LitElement {
  render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-tabs style="max-width: 100%; width: 400px;">
        <vaadin-tab>Analytics</vaadin-tab>
        <vaadin-tab>Customers</vaadin-tab>
        <vaadin-tab>Dashboards</vaadin-tab>
        <vaadin-tab>Documents</vaadin-tab>
        <vaadin-tab>Orders</vaadin-tab>
      </vaadin-tabs>
      <!-- end::snippet[] -->
    `;
  }
}
